packages/editor/src/lib/globals/menus.ts

```diff
@@ -0,0 +1,119 @@
+import { atom } from '@tldraw/state'
+
+/** @public */
+export const tlmenus = {
+	/**
+	 * A set of strings representing any open menus. When menus are open,
+	 * certain interactions will behave differently; for example, when a
+	 * draw tool is selected and a menu is open, a pointer-down will not
+	 * create a dot (because the user is probably trying to close the menu)
+	 * however a pointer-down event followed by a drag will begin drawing
+	 * a line (because the user is BOTH trying to close the menu AND start
+	 * drawing a line).
+	 *
+	 * @public
+	 */
+	menus: atom<string[]>('open menus', []),
```

Comment:
Why not a map here, similar to timers?

Reply:
could do that

Reply:
Hmm, seems easier to keep this flat. Let's come back to this later and pick a nicer structure.

Reply:
(the "front door" shouldn't change)